core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -140,23 +145,18 @@ public void sync() throws IOException {
         if (syncNeeded()) {
             synchronized (this) {
                 ensureOpen(); // this call gives a better exception that the incRef if we are closed by a tragic event
-                channelReference.incRef();
+                final long offsetToSync;
+                final int opsCounter;
+                outputStream.flush();
```

Comment:
Hmm, I think if this hits `Throwable` we should also `closeWithTragicEvent`, and we should javadoc this method that it declares tragedy on exceptions so callers don't need to?